app/views/site/_study_settings_form.html.erb

```diff
@@ -160,4 +161,11 @@
             updateForm.submit();
         });
     });
+
+    $("#update-study-settings-form").on('change', '.share-permission', function() {
+        var newPermission = $(this).val();
+        var permissionText = <%= raw StudyShare::PERMISSION_DESCRIPTION_MAP.to_json %>;
```

Comment:
`permissionText` seems to be a constant.  So it would be better to assign it once (e.g. right above this event handler) rather than every time this callback is invoked.  This is a minor structuring issue that doesn't affect performance, though, so I don't consider it blocking.

Reply:
I take the point, but we're doing the same thing in multiple places, and given that all models/constants are held in memory in Rails, this doesn't involve a query.  It's just accessing the object.  So I don't think it makes much of a substantive difference in terms of performance, and it makes it a little more complicated (one more chance for something to break).